src/sentry/api/endpoints/project_users.py

```diff
@@ -28,12 +29,28 @@ def get(self, request, project):
             pieces = request.GET["query"].strip().split(":", 1)
             if len(pieces) != 2:
                 return Response([])
-            try:
-                queryset = queryset.filter(
-                    **{f"{EventUser.attr_from_keyword(pieces[0])}__icontains": pieces[1]}
-                )
-            except KeyError:
-                return Response([])
+            # username and ip can return multiple eventuser objects
+            if any(p in "ip" or p in "username" for p in pieces):
+                try:
+                    queryset = queryset.filter(
+                        project_id=project.id,
+                        **{f"{EventUser.attr_from_keyword(pieces[0])}__icontains": pieces[1]},
+                    )
+                except EventUser.DoesNotExist:
+                    return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
Since we're using `.filter` here now we'll never have a `DoesNotExist` raised, since that only happens on `.get`. We probably don't need to catch anything here, we'll end up returning a response with no results.